# HG changeset patch
# User Sean Farley <sean.michael.farley@gmail.com>
# Date 1397847162 18000
#      Fri Apr 18 13:52:42 2014 -0500
# Node ID a47e3a6b57a431073b3e3244893e887f3aa9b1f5
# Parent  f610b440a202997768c53e526a1d1fd9e10f1c0a
make: change compilers to be variables

diff --git a/admin/Makefile.def b/admin/Makefile.def
--- a/admin/Makefile.def
+++ b/admin/Makefile.def
@@ -71,16 +71,16 @@ CLDLIBS = $(GAG_LIB_FLAGS)
 FLDLIBS = $(GAG_LIB_FLAGS)
 
 # The following variables are provided to the user so that he can override
 # the preprocessing, compilation and linking options defined by default
 # later in this file. User can modify them here.
-CPPFLAGS = $(GAG_CPPFLAGS)
-SLDFLAGS = $(GAG_SLDFLAGS)
-FLDFLAGS = $(GAG_FLDFLAGS)
-CLDFLAGS = $(GAG_CLDFLAGS)
-FFLAGS = $(GAG_FFLAGS)
-CFLAGS = $(GAG_CFLAGS)
+CPPFLAGS = @CPPFLAGS@ $(GAG_CPPFLAGS)
+SLDFLAGS = @LDFLAGS@ $(GAG_SLDFLAGS)
+FLDFLAGS = @LDFLAGS@ $(GAG_FLDFLAGS)
+CLDFLAGS = @LDFLAGS@ $(GAG_CLDFLAGS)
+FFLAGS = @FCFLAGS@ $(GAG_FFLAGS)
+CFLAGS = @CCFLAGS@ $(GAG_CFLAGS)
 
 # The following variables are for mandatory GLOBAL definitions.
 # They should be modified only in this file.
 GLOBAL_CPPFLAGS = 
 GLOBAL_SLDFLAGS = 
@@ -229,11 +229,11 @@ GLOBAL_DIRTY = $(builddir) *~ TAGS Chang
 
 ###########################################################################
 # Fortran preprocessing
 
 # Preprocessor command
-CPP = cpp
+CPP = @CPP@
 ifeq ($(GAG_OS_KIND),solaris)
   CPP = /usr/ccs/lib/cpp
 endif
 ifeq ($(GAG_OS_KIND),aix)
   CPP = /usr/ccs/lib/cpp
@@ -307,12 +307,12 @@ endif
 
 ###########################################################################
 # C compilation
 
 # C compiler name
-CC = gcc
-CXX = g++
+CC = @CC@
+CXX = @CXX@
 ifeq ($(GAG_OS_KIND),aix)
   CC = xlc
 endif
 ifeq ($(GAG_OS_KIND),osf)
   CC = cc
@@ -331,11 +331,11 @@ GLOBAL_CFLAGS += -I$(execincdir)
 ifneq ($(useincdir),)
     GLOBAL_CFLAGS += -I$(useincdir)
 endif
 
 # GCC flags
-ifeq ($(CC),gcc)
+ifeq ($(CC),@CC@)
   # Generic to GCC
   GLOBAL_CFLAGS += -pipe -fPIC
   OPTION_CFLAGS += -Wall
   ifeq ($(GAG_USE_32BITS),yes)
     GLOBAL_CFLAGS += -m32
diff --git a/admin/Makefile.main b/admin/Makefile.main
--- a/admin/Makefile.main
+++ b/admin/Makefile.main
@@ -16,11 +16,11 @@ ifeq ($(GAG_OS_KIND),mingw)
 RC_FILES = $(patsubst %,win-%.rc,$(EXECUTABLES))
 RC_EXISTING_FILES = $(wildcard $(RC_FILES))
 RC_DLLS = $(patsubst %.rc,$(libdir)/%.dll,$(RC_EXISTING_FILES))
 TARGETS += $(RC_DLLS)
 $(libdir)/%.dll: $(builddir)/%.o
-	gcc -o $@ -shared $?
+	$(CC) -o $@ -shared $?
 endif
 ifneq ($(GAG_OS_KIND),cygwin)
 DIRTY = $(PCL) $(EXECS) $(OBJECTS)
 else
 DIRTY = $(PCL) $(patsubst %,%.exe,$(EXECS)) $(OBJECTS)
